app/services/claims/submit.rb

```diff
@@ -0,0 +1,29 @@
+class Claims::Submit
+  include ServicePattern
+
+  def initialize(claim:)
+    @claim = claim
+  end
+
+  def call
+    updated_claim.save!
+  end
+
+  private
+
+  attr_reader :claim
+
+  def updated_claim
+    @updated_claim ||= begin
+      claim.draft = false
+      claim.submitted_at = Time.current
+      claim.reference = generate_reference
+      claim
+    end
+  end
+
+  def generate_reference
+    reference = SecureRandom.random_number(99_999_999) while Claim.exists?(reference:)
```

Comment:
I'm wondering if this is the _logical_ way to do this... Should we be treating the reference number as a **string** of 8 random numbers instead of a number, then converting it to a string?

```ruby
8.times.map { ("0".."9").to_a.sample }.join
```

Just a thought, but likely less optimal. No changes needed. :)